rtos/TARGET_CORTEX/TOOLCHAIN_GCC_ARM/mbed_boot_gcc_arm.c

```diff
@@ -52,24 +52,13 @@ extern void __libc_init_array(void);
  */
 void software_init_hook(void)
 {
-    unsigned char *free_start = HEAP_START;
-    uint32_t free_size = HEAP_SIZE;
-
-#ifdef ISR_STACK_START
     /* Interrupt stack explicitly specified */
-    mbed_stack_isr_size = ISR_STACK_SIZE;
-    mbed_stack_isr_start = ISR_STACK_START;
-#else
-    /* Interrupt stack -  reserve space at the end of the free block */
-    mbed_stack_isr_size = ISR_STACK_SIZE < free_size ? ISR_STACK_SIZE : free_size;
-    mbed_stack_isr_start = free_start + free_size - mbed_stack_isr_size;
-    free_size -= mbed_stack_isr_size;
-#endif
+    mbed_stack_isr_size = (unsigned char *)ISR_STACK_SIZE;
```

Comment:
If you plan to define `__HeapLimit` symbol in all gcc linker scripts I suggest to remove above blocks:
`#if !defined(ISR_STACK_SIZE)` and `#if !defined(HEAP_START)` and define
isr stack/heap start/size using linker script symbols directly.

Reply:
Linker symbols considered, but will leave ISR_STACK_SIZE / HEAP_START checks to give targets flexibility to overwrite. Might remove them in future releases.